File: it/test-basic/expected/suite/java/test/Foo.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Objects;
import java.util.Optional;

public class Foo {
  /**
   * <pre>
   * The field.
   * </pre>
   */
  @JsonProperty("field")
  private final String field;

  @JsonCreator
  public Foo(
    @JsonProperty("field") final String field
  ) {
    Objects.requireNonNull(field, "field");
    this.field = field;
  }

  /**
   * <pre>
   * The field.
   * </pre>
   */
  public String getField() {
    return this.field;
  }

  @Override
  public int hashCode() {
    int result = 1;
    result = result * 31 + this.field.hashCode();
    return result;
  }

  @Override
  public boolean equals(final Object other) {
    if (other == null) {
      return false;
    }

    if (!(other instanceof Foo)) {
      return false;
    }

    @SuppressWarnings("unchecked")
    final Foo o = (Foo) other;

    if (!this.field.equals(o.field)) {
      return false;
    }

    return true;
  }

  @Override
  public String toString() {
    final StringBuilder b = new StringBuilder();

    b.append("Foo");
    b.append("(");
    b.append("field=");
    b.append(this.field.toString());
    b.append(")");

    return b.toString();
  }

  public static class Builder {
    private Optional<String> field = Optional.empty();

    public Builder field(final String field) {
      this.field = Optional.of(field);
      return this;
    }

    public Foo build() {
      final String field = this.field.orElseThrow(() -> new RuntimeException("field: is required"));

      return new Foo(field);
    }
  }
}

File: it/test-basic/expected/suite/csharp/Test/Foo.cs
using Newtonsoft.Json;
using System;
using System.Text;

namespace Test {
  [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
  public class Foo {
    /// <summary>
    /// The field.
    /// </summary>
    [JsonProperty("field", Required = Required.DisallowNull)]
    public String field {
      get;
    }

    [JsonConstructor]
    public Foo(
      [JsonProperty("field", Required = Required.DisallowNull)] String field
    ) {
      this.field = field;
    }

    public override Int32 GetHashCode() {
      Int32 result = 1;
      result = result * 31 + this.field.GetHashCode();
      return result;
    }

    public override Boolean Equals(Object other) {
      Foo o = other as Foo;

      if (o == null) {
        return false;
      }

      if (!this.field.Equals(o.field)) {
        return false;
      }

      return true;
    }

    public override String ToString() {
      StringBuilder b = new StringBuilder();

      b.Append("Foo");
      b.Append("(");
      b.Append("field=");
      b.Append(this.field);
      b.Append(")");

      return b.ToString();
    }
  }
}
